[Merged by Bors] - feat(Analysis/RCLike/Basic): PosMulReflectLE

Timeroot opened the pull request:
Adds
```
@[simp]
theorem RCLike.inv_nonneg {x : 𝕜} : 0 ≤ x⁻¹ ↔ 0 ≤ x
```
to match [inv_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/GroupWithZero/Unbundled.html#inv_nonneg) and [Rat.inv_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Field/Rat.html#Rat.inv_nonneg).

Upstreamed from [QuantumInfo](https://github.com/Timeroot/Lean-QuantumInfo)

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

b-mehta reviewed (commented):
Aren't both of these already given by `inv_nonneg` and `inv_nonneg_of_nonneg` in the general case?

Timeroot commented:
> Aren't both of these already given by inv_nonneg and inv_nonneg_of_nonneg in the general case?

Oops, you're right - kind of! I saw they didn't apply which is why I had written new ones. But the "real" problem was that `PosMulReflectLE`, the instance that they need, has never been provided for RCLike. (There is `PosMulStrictMono`, which is sufficient to synthesize `PosMulReflectLE` if the ring is a linear order; but ComplexOrder isn't a linear order.)

So I've removed the `RCLike.inv_nonneg` and added a `PosMulReflectLE` instance now.

b-mehta commented on Mathlib/Analysis/RCLike/Basic.lean:
```diff
@@ -879,6 +879,19 @@ theorem ofReal_mul_neg_iff (x : ℝ) (z : K) :
+lemma instPosMulReflectLE : PosMulReflectLE K := by
+  constructor
+  intro a b c (h : _ * _ ≤ _ * _)
+  obtain ⟨a',ha1,ha2⟩ := pos_iff_exists_ofReal.mp a.2
```
```suggestion
  obtain ⟨a', ha1, ha2⟩ := pos_iff_exists_ofReal.mp a.2
```

b-mehta commented on Mathlib/Analysis/RCLike/Basic.lean:
```diff
@@ -879,6 +879,19 @@ theorem ofReal_mul_neg_iff (x : ℝ) (z : K) :
+  obtain ⟨a',ha1,ha2⟩ := pos_iff_exists_ofReal.mp a.2
+  rw [← sub_nonneg]
+  rw [← ha2, ← sub_nonneg, ← mul_sub, le_iff_lt_or_eq] at h
+  rcases h with h|h
```
```suggestion
  rcases h with h | h
```

b-mehta commented on Mathlib/Analysis/RCLike/Basic.lean:
```diff
@@ -879,6 +879,19 @@ theorem ofReal_mul_neg_iff (x : ℝ) (z : K) :
+  rcases h with h|h
+  · rw [ofReal_mul_pos_iff] at h
+    classical exact le_of_lt <| h.by_cases (False.elim <| not_lt_of_gt ·.1 ha1) (·.2)
+  · exact le_of_eq ((mul_eq_zero_iff_left <| ofReal_ne_zero.mpr ha1.ne').mp (h.symm)).symm
```
```suggestion
  · exact ((mul_eq_zero_iff_left <| ofReal_ne_zero.mpr ha1.ne').mp h.symm).ge
```
untested, but I think this should work?

b-mehta reviewed (commented):
bors d+
